datalad/distribution/update.py

```diff
@@ -355,73 +443,123 @@ def _choose_merge_target(repo, branch, remote, cfg_remote):
 
     Returns
     -------
-    str (the merge target) or None if a choice wasn't made.
+    str (the target) or None if a choice wasn't made.
     """
-    merge_target = None
+    target = None
     if cfg_remote and remote == cfg_remote:
-        # Use the configured cfg_remote branch as the merge target.
+        # Use the configured cfg_remote branch as the target.
         #
-        # In this scenario, it's tempting to use FETCH_HEAD as the merge
-        # target. That would be the equivalent of 'git pull REMOTE'. But doing
+        # In this scenario, it's tempting to use FETCH_HEAD as the target. For
+        # a merge, that would be the equivalent of 'git pull REMOTE'. But doing
         # so would be problematic when the GitRepo.fetch() call was passed
         # all_=True. Given we can't use FETCH_HEAD, it's tempting to use the
         # branch.*.merge value, but that assumes a value for remote.*.fetch.
-        merge_target = repo.call_git_oneline(
+        target = repo.call_git_oneline(
             ["rev-parse", "--symbolic-full-name", "--abbrev-ref=strict",
              "@{upstream}"],
             read_only=True)
     elif branch:
         remote_branch = "{}/{}".format(remote, branch)
         if repo.commit_exists(remote_branch):
-            merge_target = remote_branch
-    return merge_target
+            target = remote_branch
+    return target
 
 
-#  Merge functions
+#  Update functions
 
 
-def _choose_merge_fn(repo, is_annex=False, adjusted=False):
-    if adjusted and is_annex:
-        # For adjusted repos, blindly sync.
-        merge_fn = _annex_sync
-    elif is_annex:
-        merge_fn = _annex_plain_merge
-    elif adjusted:
+def _choose_update_fn(repo, how, is_annex=False, adjusted=False):
+    if adjusted and how != "merge":
         raise RuntimeError(
-            "bug: Upstream checks should make it impossible for "
-            "adjusted=True, is_annex=False")
+            "bug: Upstream checks should abort if adjusted is used "
+            "with action other than 'merge'")
+    elif how in ["merge", "ff-only"]:
+        if adjusted and is_annex:
+            # For adjusted repos, blindly sync.
+            fn = _annex_sync
+        elif is_annex:
+            fn = _annex_plain_merge
+        elif adjusted:
+            raise RuntimeError(
+                "bug: Upstream checks should make it impossible for "
+                "adjusted=True, is_annex=False")
+        else:
+            fn = _plain_merge
+    elif how == "reset":
+        fn = _reset_hard
+    elif how == "checkout":
+        fn = _checkout
     else:
-        merge_fn = _plain_merge
-    return merge_fn
+        raise ValueError(f"Unrecognized value for `how`: {how}")
```

Comment:
why did I think we still can't use f-strings...?  we do require python >= 3.6 and f-strings were added there... verified that this worked in 3.6.8rc1. so all good and I will use them now!